test/controllers/changePref/EmailToUseControllerSpec.scala

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.changePref
+
+import assets.CustomerInfoConstants.fullCustomerInfoModel
+import common.SessionKeys
+import connectors.httpParsers.GetCustomerInfoHttpParser.GetCustomerInfoResponse
+import controllers.ControllerBaseSpec
+import forms.YesNoForm.yesNo
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsEmpty, AnyContentAsFormUrlEncoded}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.changePref.EmailToUseView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EmailToUseControllerSpec extends ControllerBaseSpec {
+
+  def setup(result: GetCustomerInfoResponse): Any =
+    when(mockVatSubscriptionService.getCustomerInfo(any[String])(any[HeaderCarrier], any[ExecutionContext]))
+      .thenReturn(Future.successful(result))
+
+  val testValidationEmail: String = "validation@example.com"
+
+  lazy val existingEmailSessionRequest: FakeRequest[AnyContentAsEmpty.type] =
+    request.withSession(SessionKeys.validationEmailKey -> testValidationEmail)
+
+  lazy val noEmailSessionRequest: FakeRequest[AnyContentAsEmpty.type] =
+    request
+
+  val view: EmailToUseView = injector.instanceOf[EmailToUseView]
+
+  def target(result: GetCustomerInfoResponse = Right(fullCustomerInfoModel)): EmailToUseController = {
+    setup(result)
+
+    new EmailToUseController(
+      mockVatSubscriptionService,
+      mockErrorHandler,
+      view
+    )
+  }
+
+  "Calling the show action in EmailToUseController" when {
+
+    "the letterToConfirmedEmail switch is enabled" when {
+
+      "there is an email in session" should {
```

Comment:
Could you put an extra test here and in the scenario below to confirm that the two session keys are populated